lib/multipass_ex/coding.ex

```diff
@@ -0,0 +1,56 @@
+defmodule MultipassEx.Coding do
+  @moduledoc """
+  Encoding / Decoding of Multipass data.
+
+  The MultipassEx gem truncates padding characters from base64 encoding whether
+  they are present or not. It then adds a single padding character to the end
+  of the base64 encoded string before decoding without reguard to its necessity.
+  This causes issues if you decode without `padding: false` as it causes padding
+  errors. Encoding should also be done without padding.
+
+  The gem also replaces characters after base64 encoding to make the string
+  "url safe". This happens before base64 decoding and after base64 encoding.
+  """
+
+  alias MultipassEx.DecodingError
+
+  @doc """
+  Base64 encode the string without padding and replace characters to match the
+  ruby gems "url_safe" flag.
+  """
+  @spec encode(binary) :: binary
+  def encode(data) do
+    data
+    |> Base.encode64(padding: false)
+    |> String.replace("+", "-", global: true)
+    |> String.replace("/", "_", global: true)
+    |> String.replace("\n", "", global: true)
```

Comment:
`global: true` is the default, but I understand if you want to be explicit.

Reply:
TIL ! Thanks